### PR TITLE
worker: make run as a default method

### PR DESCRIPTION
**Pull request by overvenus**
Signed-off-by: Neil Shen <overvenus@gmail.com>

## What have you changed? (mandatory)

This PR makes `Runable::run` as a default method, so users can skip unnecessary impl run. 

## What are the type of the changes? (mandatory)

- Improvement (non-breaking change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)

Unit tests.

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)

No.

## Does this PR affect tidb-ansible update? (mandatory)

No.

## Refer to a related PR or issue link (optional)

https://github.com/tikv/tikv/pull/3551#discussion_r216543627

### CODE REVIEW

File: src/util/worker/mod.rs [hunk 1]
```diff
@@ -84,7 +84,9 @@ impl<T> Debug for ScheduleError<T> {
 
 pub trait Runnable<T: Display> {
     /// Run a task.
-    fn run(&mut self, t: T);
+    fn run(&mut self, _: T) {
```
Comment (BusyJay) [inline src/util/worker/mod.rs]: Why not remove it?

Comment (overvenus) [reply src/util/worker/mod.rs]: There are some workers impls `run`.

**Comment by overvenus** [comment]
The endpoint worker :) @breeswish 

https://github.com/tikv/tikv/pull/3593/files#diff-49763d722eeb4620a599e6bddecf8f56L608

**Comment by overvenus** [comment]
PTAL, thanks

File: src/coprocessor/endpoint.rs [hunk 2]
```diff
@@ -364,10 +364,6 @@ impl Display for RequestTask {
 
 impl<E: Engine> Runnable<Task<E>> for Host<E> {
     // TODO: limit pending reqs
```
Comment (Connor1996) [inline src/coprocessor/endpoint.rs]: remove TODO

Comment (overvenus) [reply src/coprocessor/endpoint.rs]: I dont think it is resolved.

**Pull request merged**